components/uniset/src/lib.rs

```diff
@@ -69,6 +70,12 @@ pub enum UnicodeSetError {
     InvalidRange(u32, u32),
 }
 
+impl fmt::Display for UnicodeSetError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Error with UnicodeSet: {:?}", self)
```

Comment:
Optional: if you implement fmt::Display, either present the user with nice output, or just fully delegate to fmt::Debug.  The string form of this error will be something like: "Error with UnicodeSet: UnicodeSetError{InvalidRange(5, 3)}".  You should either remove the "Error with UnicodeSet" or make the message nicer, like "Error with UnicodeSet: Invalid range: 5, 3"

Reply:
Done.